include/dap/any.h

```diff
@@ -143,6 +143,7 @@ any& any::operator=(const T& val) {
     alloc(type->size(), type->alignment());
     type->copyConstruct(value, &val);
   } else {
+    assert(value != nullptr);
```

Comment:
Is there a way we can tell clang-analyzer this is not null without introducing runtime overhead? (not that cppdap is massively performance critical)